src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs

```diff
@@ -194,19 +194,52 @@ public override IEnumerable<TypeInfo> DefinedTypes
 
         public override bool IsCollectible => GetIsCollectible(GetNativeHandle());
 
+        // GetResource will return a pointer to the resources in memory.
+        [DllImport(JitHelpers.QCall, CharSet = CharSet.Unicode)]
+        private static extern unsafe byte* GetResource(RuntimeAssembly assembly,
+                                                       string resourceName,
+                                                       out ulong length);
+
         // Load a resource based on the NameSpace of the type.
-        [System.Security.DynamicSecurityMethod] // Methods containing StackCrawlMark local var has to be marked DynamicSecurityMethod
         public override Stream GetManifestResourceStream(Type type, string name)
         {
-            StackCrawlMark stackMark = StackCrawlMark.LookForMyCaller;
-            return GetManifestResourceStream(type, name, false, ref stackMark);
+            StringBuilder sb = new StringBuilder();
+            if (type == null)
+            {
+                if (name == null)
+                    throw new ArgumentNullException(nameof(type));
+            }
+            else
+            {
+                string nameSpace = type.Namespace;
+                if (nameSpace != null)
+                {
+                    sb.Append(nameSpace);
+                    if (name != null)
+                        sb.Append(Type.Delimiter);
+                }
+            }
+
+            if (name != null)
+                sb.Append(name);
+
+            return GetManifestResourceStream(sb.ToString());
         }
 
-        [System.Security.DynamicSecurityMethod] // Methods containing StackCrawlMark local var has to be marked DynamicSecurityMethod
-        public override Stream GetManifestResourceStream(string name)
+        public unsafe override Stream GetManifestResourceStream(string name)
         {
-            StackCrawlMark stackMark = StackCrawlMark.LookForMyCaller;
-            return GetManifestResourceStream(name, ref stackMark, false);
+            ulong length = 0;
```

Comment:
Nit: You can change this to `uint` to avoid the cast from uint to ulong and back.